### ISSUE TITLE
Support for auth proxy authentication

### ISSUE THREAD

**Issue by l0tzi** [panodata/grafana-client#31]
I am using the  auth proxy authentication to access Grafana with an apache webserver infront of Grafana (documetation: [configure auth proxy authentication](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/auth-proxy/#configure-auth-proxy-authentication)).

Support for this kind of auth would be very nice. I think it could be quite easy to implement this feature.  I can create a pull request if it is desired.

BR

**Comment by amotl** [comment]
Dear @l0tzi,

thank you for writing in.

As far as I understand the situation, this procedure will shift the authenticator role from Grafana to Apache, where it would still use the same authentication mechanics, namely [HTTP Basic Auth](https://en.wikipedia.org/wiki/Basic_access_authentication), as outlined at [1].

In this manner, the existing code for `grafana-client` to use the HTTP Basic Authentication method [2] should work out of the box, no?

https://github.com/panodata/grafana-client/blob/dce5a301e6d6820f1d27832c2534ec0365ae49be/grafana_client/client.py#L100-L104

Maybe you can explain a bit why `grafana-client` would need to support a different authentication method then, and which method this should be?

With kind regards,
Andreas.

[1] https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/auth-proxy/#making-apaches-auth-work-together-with-grafanas-authproxy
[2] https://github.com/panodata/grafana-client#authentication

**Comment by l0tzi** [comment]
Dear @amotl,
in my case we are using kerberos to autenticate the AD users, which are using Grafana. Our script to configure Grafan is running local on the server, so we connect directly to the grafana instance without passing apache. In this case it's only necessary to set `X-WEBAUTH-USER`  of the request header:
```
            request.headers.update({'X-WEBAUTH-USER': user})
```
This is enough to authenticate to grafana in our case.

Kind Regards
Lorenz

**Comment by amotl** [comment]
Dear Lorenz,

thank you for explaining your scenario. Now it makes sense to me. A generic `HeaderAuth` mechanism will probably be the most flexible way of supporting your use case. What do you think?

With kind regards,
Andreas.

**Comment by amotl** [comment]
Dear Lorenz,

thanks again for your suggestion. It got implemented with #37 and will be part of the next release.

With kind regards,
Andreas.

**Comment by amotl** [comment]
Hi again,

do you think the current implementation will be helpful? Let us know if it works for you.

With kind regards,
Andreas.
